lib/api/pokedex_api_spec/handler/pokemon_handler.dart

```diff
@@ -1,21 +1,25 @@
 import 'package:pokedex/api/api_client.dart';
 import 'package:pokedex/api/pokedex_api_spec/model/pokemon_model.dart';
+import 'package:pokedex/utils/strings.dart' as str;
 
 class PokemonApi {
   final ApiClient apiClient;
 
   PokemonApi(this.apiClient);
 
-  static const int _limitAmount = 20;
-
-  Future<Pokemon> getPokemonList() async {
+  Future<List<Pokemon>> getPokemonList(String? offset, String? limit) async {
     final queryParams = <String, dynamic>{};
+    if (limit != null) {
+      queryParams['limit'] = limit;
+    }
+    if (offset != null) {
+      queryParams['offset'] = offset;
+    }
 
     final baseUri = Uri.parse(apiClient.dio.options.baseUrl);
-    final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/pokemon/?limit=$_limitAmount');
-
-    return apiClient.dio
+    final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/${str.pokemonRoute}/');
```

Comment:
no need for the end `/` you aren't gonna trigger a new path here right? 🙂

Reply:
I have removed it yes no more new path will be triggered here